iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveSubstationTest.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.test.AbstractConverterTest;
+import com.powsybl.iidm.network.DefaultNetworkListener;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.test.EurostagTutorialExample1Factory;
+import com.powsybl.iidm.network.test.FourSubstationsNodeBreakerFactory;
+import com.powsybl.iidm.network.test.HvdcTestNetwork;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Maissa Souissi <maissa.souissi at rte-france.com> eurostag-remove-voltage-level-bb.xml
+ */
+class RemoveSubstationTest extends AbstractConverterTest {
+    private final Set<String> removedObjects = new HashSet<>();
+    private final Set<String> beforeRemovalObjects = new HashSet<>();
+
+    @AfterEach
+    public void tearDown() {
+        removedObjects.clear();
+    }
+
+    private void addListener(Network network) {
```

Comment:
Why we use a `beforeRemovalObjects` ?
It'is always the same as `afterRemoval` ?